client/src/app/employee/employee-detail/state/employee.service.ts

```diff
@@ -0,0 +1,68 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { ToastrService } from 'ngx-toastr';
+import { throwError } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+import { EnvService } from 'src/app/shared/env.service';
+import { Employee, EmployeeUpdate } from './employee.model';
+import { EmployeeStore } from './employee.store';
+
+@Injectable({ providedIn: 'root' })
+export class EmployeeService {
+  readonly apiRootUrl: string;
+
+  constructor(
+    envService: EnvService,
+    private http: HttpClient,
+    private store: EmployeeStore,
+    private toastr: ToastrService,
+  ) {
+    this.apiRootUrl = envService.apiRootUrl;
+  }
+
+  getEmployee(id: string) {
+    this.store.setLoading(true);
+
+    this.http
+      .get<Employee>(`${this.apiRootUrl}/Employees/${id}`)
+      .pipe(
+        catchError((err) => {
+          this.toastr.error(`Could not get employee ${id}`);
+          return throwError(err);
```

Comment:
If the employee request fails, what will currently happen is the error will not be propagated anywhere.

Since the subscription is happening below and there is no `error` callback to the Observer passed to `subscribe()`, the `return throwError(err)` doesn't go anywhere.

It's probably more appropriate to do the `this.toast.error(...)` in the `error` callback of the Observer below.

Reply:
fixed, i believe writing all the `error`, `next`, and `complete` code in `subscribe()` means i could get rid of the `pipe()` completely.

Reply:
Yup, in fact the only time you really need `.pipe` is when you don't subscribe.